packages/messaging/src/controllers/sw-controller.ts

```diff
@@ -206,9 +206,9 @@ export class SwController implements FirebaseMessaging, FirebaseService {
       const payload = externalizePayload(internalPayload);
 
       if (typeof this.bgMessageHandler === 'function') {
-        this.bgMessageHandler(payload);
+        await this.bgMessageHandler(payload);
       } else {
-        this.bgMessageHandler.next(payload);
+        await this.bgMessageHandler.next(payload);
```

Comment:
This is not compatible with `Observer` signature where `next()` should return `void`. Perhaps we should only take a callback instead of an observer in `onBackgroundMessage()` ?